lib/connection/connection.js

```diff
@@ -30,13 +31,15 @@ const JsonCredentialManager = require('../authentication/secure_storage/json_cre
  */
 function Connection(context) {
   // validate input
+  Logger.getInstance().debug('Connection object is being constructed');
   Errors.assertInternal(Util.isObject(context));
 
   const services = context.getServices();
   const connectionConfig = context.getConnectionConfig();
```

Comment:
Also, it would be feasible to show parameters which are used to open the connection, similar to JDBC driver:
``` java
 logger.info(
        "Opening session with server: {}, account: {}, user: {}, password is {}, role: {}, database: {}, schema: {},"
            + " warehouse: {}, validate default parameters: {}, authenticator: {}, ocsp mode: {},"
            + " passcode in password: {}, passcode is {}, private key is {}, disable socks proxy: {},"
            + " application: {}, app id: {}, app version: {}, login timeout: {}, retry timeout: {}, network timeout: {},"
            + " query timeout: {}, connection timeout: {}, socket timeout: {}, tracing: {}, private key file: {}, private key pwd is {},"
            + " enable_diagnostics: {}, diagnostics_allowlist_path: {},"
            + " session parameters: client store temporary credential: {}, gzip disabled: {}, browser response timeout: {}",
(...)
```

Reply:
Sure, thanks :) 
I passed the password as well since it is masked later on using SecretDetector

Reply:
Let's not pass it, but you can create a function similar to the JDBC drivers' [SFLoggerUtil.isVariableProvided](https://github.com/snowflakedb/snowflake-jdbc/blob/9b7c9eb3e02c93cf7788c9866e1a483fe978e1be/src/main/java/net/snowflake/client/log/SFLoggerUtil.java#L39). It is safer to do it this way, even if we have a secret detector that scans through afterwards.

Reply:
Same for passcode, private key, private key pass.

Reply:
Sure, should be fixed now 👍